test/equivalences.jl

```diff
@@ -23,92 +23,66 @@
 
         gpr_post = posterior(fx, y) # Exact GP regression
         vfe_post = approx_posterior(VFE(), fx, y, fz) # Titsias posterior
-        svgp_post = approx_posterior(SVGP(), fz, q_ex) # Hensman (2013) exact posterior
+        svgp_post = posterior(SVGP(fz, q_ex)) # Hensman (2013) exact posterior
 
         @test mean(gpr_post, x) ≈ mean(svgp_post, x) atol = 1e-10
         @test cov(gpr_post, x) ≈ cov(svgp_post, x) atol = 1e-10
 
         @test mean(vfe_post, x) ≈ mean(svgp_post, x) atol = 1e-10
         @test cov(vfe_post, x) ≈ cov(svgp_post, x) atol = 1e-10
 
-        @test elbo(fx, y, fz, q_ex) ≈ logpdf(fx, y) atol = 1e-6
+        @test elbo(SVGP(fz, q_ex), fx, y) ≈ logpdf(fx, y) atol = 1e-6
     end
 
     @testset "optimised posterior" begin
         jitter = 1e-5
 
         make_gp(kernel) = GP(kernel)
 
-        ## FIRST - define the models
-        # GPR - Exact GP regression
-        struct GPRModel
-            k # kernel parameters
-        end
-        Flux.@functor GPRModel
-
-        function (m::GPRModel)(x)
-            f = make_gp(make_kernel(m.k))
-            fx = f(x, lik_noise)
-            return fx
-        end
-
-        # SVGP - Sparse variational GP regression (Hensman 2014)
+        # SVGP model
         struct SVGPModel
             k # kernel parameters
             z # inducing points
             m # variational mean
             A # variational covariance sqrt (Σ = A'A)
         end
-        Flux.@functor SVGPModel (k, m, A) # Don't train the inducing inputs
+        Flux.@functor SVGPModel (m, A) # Only train the variational parameters
 
-        function (m::SVGPModel)(x)
+        function construct_parts(m::SVGPModel, x)
             f = make_gp(make_kernel(m.k))
             q = MvNormal(m.m, m.A'm.A)
```

Comment:
The problem with using `MvNormal` is that it is highly suboptimal.
Right now we are computing `m.A' m.A`, but then a `cholesky` is going to be applied on it.
Can we just pass a `LowerTriangular` and directly construc the cholesky via `MvNormal(m.m, PDMat(Cholesky(m.A, 'L', 0)))` ?

Reply:
Of course the best solution would be to have our own `AbstractMvNormal` parametrized by m and A